### PR TITLE
README: update main branch references to gz-math8

### PR DESCRIPTION
**Pull request by scpeters**
# 🦟 Bug fix

Part of https://github.com/gazebo-tooling/release-tools/issues/1092

## Summary

Update branch references in the README from `main` to `gz-math8` including the jenkins badges and some markdown links.

## Checklist
- [X] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [X] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [X] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -82,12 +82,12 @@ Please see the
 # Code of Conduct
 
 Please see
-[CODE_OF_CONDUCT.md](https://github.com/gazebosim/gz-sim/blob/main/CODE_OF_CONDUCT.md).
+[CODE_OF_CONDUCT.md](https://github.com/gazebosim/gz-sim/blob/gz-math8/CODE_OF_CONDUCT.md).
```
[the remark this answers is not in the document]
Comment (scpeters) [reply README.md]: reverted in https://github.com/gazebosim/gz-math/pull/622/commits/713bf6b62348bb991b08bf92fc38dd5ee7fd884b

File: README.md [hunk 2]
```diff
@@ -9,10 +9,10 @@
 
 Build | Status
 -- | --
-Test coverage | [![codecov](https://codecov.io/gh/gazebosim/gz-math/branch/main/graph/badge.svg)](https://codecov.io/gh/gazebosim/gz-math/branch/main)
-Ubuntu Noble  | [![Build Status](https://build.osrfoundation.org/buildStatus/icon?job=gz_math-ci-main-noble-amd64)](https://build.osrfoundation.org/job/gz_math-ci-main-noble-amd64)
-Homebrew      | [![Build Status](https://build.osrfoundation.org/buildStatus/icon?job=gz_math-ci-main-homebrew-amd64)](https://build.osrfoundation.org/job/gz_math-ci-main-homebrew-amd64)
-Windows       | [![Build Status](https://build.osrfoundation.org/buildStatus/icon?job=gz_math-main-win)](https://build.osrfoundation.org/job/gz_math-main-win)
+Test coverage | [![codecov](https://codecov.io/gh/gazebosim/gz-math/branch/gz-math8/graph/badge.svg)](https://codecov.io/gh/gazebosim/gz-math/branch/gz-math8)
```
[the remark this answers is not in the document]
Comment (scpeters) [reply README.md]: https://github.com/gazebosim/gz-math/pull/622/commits/713bf6b62348bb991b08bf92fc38dd5ee7fd884b

**Review by iche033 — APPROVED** [review]

**Comment by scpeters** [comment]
ABI checker was failing because we still had some ignition header files in the `eigen3` component 😮 

removed in https://github.com/gazebosim/gz-math/pull/622/commits/7e0fb542f36209abf479b4c545a6c25a53e97406

**Comment by scpeters** [comment]
ABI checker is still failing due to the ignition headers on the base branch

**Comment by scpeters** [comment]
> ABI checker was failing because we still had some ignition header files in the `eigen3` component 😮
> 
> removed in [7e0fb54](https://github.com/gazebosim/gz-math/commit/7e0fb542f36209abf479b4c545a6c25a53e97406)

I split this out into https://github.com/gazebosim/gz-math/pull/624 since it's making the ABI checker fail. I'll plan to merge that and then this branch should have clean CI (want to make sure the ABI checker works well on noble)

**Pull request merged**